sirepo/http_request.py

```diff
@@ -27,7 +29,8 @@ def init(**imports):
 
 
 def is_spider():
-    return user_agents.parse(flask.request.headers.get('User-Agent')).is_bot
+    a = flask.request.headers.get('User-Agent')
+    return user_agents.parse(a).is_bot or any([a in b or b in a for b in _BOTS])
```

Comment:
This code is unclear: `a in b or b in a for b in _BOTS`. It looks as if you don't actually know what's going on. I do not think this needs to be generalized with a constant. It's a special case due to a missing definition in user_agents. For this specific case, it's better to document a solution:

```py
if 'python-requests' in a:
    # user_agent doesn't see Python's requests module as a bot.
    # The package robot_detection does see it, but we don't want to introduce another dependency.
    return True
return user_agents.parse(a).is_bot
```

Reply:
db24678